modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ImageCardTag.java

```diff
@@ -32,314 +27,17 @@
 	public int doStartTag() throws JspException {
 		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
 
-		return super.doStartTag();
-	}
-
-	@Override
-	public List<DropdownItem> getActionDropdownItems() {
-		List<DropdownItem> actionDropdownItems = super.getActionDropdownItems();
-
-		if ((actionDropdownItems == null) && (_imageCard != null)) {
-			return _imageCard.getActionDropdownItems();
-		}
-
-		return actionDropdownItems;
-	}
-
-	/**
-	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
-	 */
-	@Deprecated
-	@Override
-	public String getGroupName() {
-		String groupName = super.getGroupName();
-
-		if ((groupName == null) && (_imageCard != null)) {
-			return _imageCard.getGroupName();
-		}
-
-		return groupName;
-	}
-
-	@Override
-	public String getHref() {
-		String href = super.getHref();
-
-		if ((href == null) && (_imageCard != null)) {
-			return _imageCard.getHref();
-		}
-
-		return href;
-	}
-
-	@Override
-	public String getIcon() {
-		String icon = super.getIcon();
-
-		if ((icon == null) && (_imageCard != null)) {
-			return _imageCard.getIcon();
-		}
-
-		return icon;
-	}
-
-	@Override
-	public String getImageAlt() {
-		String imageAlt = super.getImageAlt();
-
-		if ((imageAlt == null) && (_imageCard != null)) {
-			return _imageCard.getImageAlt();
+		if (getIcon() == null) {
+			setIcon("document-image");
```

Comment:
[In HorizontalCardTag](https://github.com/liferay-frontend/liferay-portal/pull/498/commits/5f6d94054cc551d88b28eb18dcf4f41457313ea6#diff-a45f3b1f2a6d6fad307b929f02e524e2759fe240dd3bf32d0b73dd5b88035225R60) we are doing the same thing, setting the default, but in getter. Is there are reason why we would need to do it here in `doStartTag`?

Reply:
Yep, `ImageCardTag` extends from `VerticalCardTag`, which defines a different default. If we place that logic in the getter you will always get in `ImageCardTag` the value defined in its parent.